### PR TITLE
Update Perfumator version to 0.4.0

### PR DESCRIPTION
**Pull request by chrisknedl**
Will remove draft status when the Perfumator version (related PR: https://github.com/se2p/perfumator/pull/2) actually gets increased to `0.3.2`. That's also why the tests currently fail.

### CODE REVIEW

File: base/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,14 @@
 FROM docker.io/library/maven:3-eclipse-temurin-17 AS perfumator-builder
 
-ARG PERFUMATOR_VERSION=0.3.0
-ENV PERFUMATOR_DOWNLOAD_URL=https://git.fim.uni-passau.de/silbereise/perfumator-java/-/archive/v${PERFUMATOR_VERSION}/perfumator-java-v${PERFUMATOR_VERSION}.zip
+ARG PERFUMATOR_VERSION=0.3.2
+ENV PERFUMATOR_DOWNLOAD_URL=https://github.com/se2p/perfumator/archive/refs/heads/main.zip
```
Comment (b-fein) [inline base/Dockerfile:4]: This will always download the latest state of the `main` branch rather than an actual version. Therefore, the behaviour of the perfumator could change without warning when the images are re-built due to unrelated changes.

I would recommend that the new version is Git-tagged in se2p/perfumator and that version is used instead.

**Review by b-fein — CHANGES_REQUESTED** [review]

File: base/Dockerfile (hunk 1 above)
Comment (chrisknedl) [reply base/Dockerfile:4]: Good idea, will tag it as soon as it's merged.

**Review by b-fein — APPROVED** [review]
